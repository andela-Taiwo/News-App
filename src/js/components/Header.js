import React from 'react';
import { Link, HashRouter } from 'react-router-dom';
import Saved from './Save';
class Header extends React.Component {
  constructor() {
    super();
    this.state = {
      authenticated: false
    }
    this.logout = this.logout.bind(this);
  }

  logout() {
    localStorage.removeItem('userProfile');
    location.reload();
  }
  navigate(){
  
    this.props.history.push('/sources');

  }

    render() {
        return (
            <nav>
              <HashRouter>
              <div className="nav-wrapper">  
                <div className="navbar-header">
                  <a href="" className="brand-logo" to="/">Articles Hub</a>
                </div>
                  <ul id="nav-mobile" className="right hide-on-med-and-down">
                    <li  onClick={this.navigate.bind(this)}><Link to="/sources">Latest News</Link></li>
                    <li>  <Link to="/archives">archives</Link></li>
                    <li>  <Link to="/saved" >Saved</Link></li>
                    <li ><Link to="/" onClick={this.logout}>Logout</Link></li>

                  
                  </ul>
            </div>
            </HashRouter>
          </nav>
        );
    }
}

export default Header;