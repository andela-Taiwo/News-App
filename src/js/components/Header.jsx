import React from 'react';
import { Link, HashRouter } from 'react-router-dom';
import Authentication from '../../helpers/Auth';

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      authed: true,
    };
    this.logout = this.logout.bind(this);
  }

  componentDidMount() {
    $('.button-collapse').sideNav();
  }
  /**
   *Logout out the user
   * set the state authentictaion
   * @memberof
   */
  logout() {
    if (this.state.authed) {
      Authentication.signOut();
      this.setState({
        authed: false,
        user: ''
      });
    }
  }
  /**
   *
   *
   * @returns Header  renders UI  for the Header component
   * @memberof Header  Header  renders UI  for the Header component
   */

  render() {
    return (
            <nav>
              <HashRouter>
                <div className="">
                  <div className="navBar">
                    <div className="">
                      <Link to="/" className="brand-logo center">
                        Articles Hub</Link>
                    </div>
                    <a href="#" data-activates="mobile-demo"
                      className="button-collapse">
                      <i className="material-icons">toc</i>
                    </a>
                    <ul id="nav-mobile" className="left hide-on-med-and-down">
                      <li><Link to="/sources" >Sources</Link></li>
                    </ul>
                    <ul id="nav-mobile" className="right hide-on-med-and-down">
                        <li>
                          <a href="#">{this.props.name}</a>
                        </li>
                      <li ><Link to="/"
                          onClick={this.logout}>Logout</Link>
                      </li>
                    </ul>
                  </div>
                  <div>
                    <ul
                        className="side-nav" id="mobile-demo"
                        style ={{
                          color: '#6D29C5',
                          fontSize: '16px',
                          marginLeft: '3px',
                          padding: '3px 5px',
                          edge: 'right', // Choose the horizontal origin
                          closeOnClick: true, // Closes side-nav on <a> clicks, useful for Angular/Meteor
                          draggable: true,
                        }}
                        >
                      <li
                        style ={{
                          color: '#6D29C5',
                          fontSize: '15px',
                          marginLeft: '3px',
                          padding: '3px 5px',
                        }}
                      ><i className="material-icons">person_pi</i>{this.props.name}</li>
                      <li ><Link to="/" onClick={this.logout}>Logout</Link></li>
                      <li><Link to="/sources" >Sources</Link></li>

                    </ul>
                  </div>
                </div>
              </HashRouter>
          </nav>
    );
  }
}

export default Header;
