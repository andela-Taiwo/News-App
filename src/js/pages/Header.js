import React from 'react';
import { Link, HashRouter } from 'react-router-dom';
import Saved from './Save';
class Header extends React.Component {
  navigate(){
    //this.props.history.pushState(null, '/');
    this.props.history.push('/');

  }

    render() {
        return (
            <nav className="nav-wrapper">
              <HashRouter>
              <div className="container-fluid">
                <div className="navbar-header">
                  <a href="" className="brand-logo" to="/">Articles Hub</a>
                </div>
                <ul id="nav-mobile" className="right hide-on-med-and-down">
                  <li className="active" onClick={this.navigate.bind(this)}><Link to="/latest">Latest News</Link></li>
                <li>  <Link to="/archives">archives</Link></li>
                <li>  <Link to="/saved"  activeClassName="active" component={Saved}>Saved</Link></li>
                  <li><input type="text"  onChange={this.handleChange} /></li>
                  <li ><Link to="/login">Login</Link></li>

                
              </ul>
            </div>
            </HashRouter>
          </nav>
        );
    }
}

export default Header;