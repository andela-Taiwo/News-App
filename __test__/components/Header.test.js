import React from 'react';
import { Link, } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { shallow, render } from 'enzyme';
import localStorageMock from './../__mocks__/LocalStorageMock';
import Header from '../../src/js/components/Header.jsx';


window.localStorage = localStorageMock;

describe('Header', () => {
  beforeEach(() => {
    window.localStorage.setItem('userProfile', JSON.stringify({
      name: 'Taiwo Memunat',
      email: 'sokunbitaiwo82@gmail.com'
    }));
  });
  it('Render without crashing', () => {
    const div = document.createElement('div');
    render(<Header />, div);
  });
});

describe('Header', () => {
  it('Found Link', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper.find('Link').length).toEqual(2);
  });
});

describe('Header (Snapshot)', () => {
  it('Header renders Articles Hub', () => {
    const component = renderer.create(<Header />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('Header (Snapshot)', () => {
  it('Header render Logout', () => {
    const component = renderer.create(<Header />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
