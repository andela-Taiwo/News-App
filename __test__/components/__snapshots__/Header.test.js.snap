// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Links  Should display Articles Hub 1`] = `
<nav>
  <div
    className=""
  >
    <div
      className="navBar"
    >
      <div
        className=""
      >
        <a
          className="brand-logo center"
          href="#/"
          onClick={[Function]}
        >
          Articles Hub
        </a>
      </div>
      <a
        className="button-collapse"
        data-activates="mobile-demo"
        href="#"
      >
        <i
          className="material-icons"
        >
          toc
        </i>
      </a>
      <ul
        className="left hide-on-med-and-down"
        id="nav-mobile"
      >
        <li>
          <a
            href="#/sources"
            onClick={[Function]}
          >
            Sources
          </a>
        </li>
      </ul>
      <ul
        className="right hide-on-med-and-down"
        id="nav-mobile"
      >
        <li>
          <a
            href="#"
          />
        </li>
        <li>
          <a
            href="#/"
            onClick={[Function]}
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
    <div>
      <ul
        className="side-nav"
        id="mobile-demo"
        style={
          Object {
            "closeOnClick": true,
            "color": "#6D29C5",
            "draggable": true,
            "edge": "right",
            "fontSize": "16px",
            "marginLeft": "3px",
            "padding": "3px 5px",
          }
        }
      >
        <li
          style={
            Object {
              "color": "#6D29C5",
              "fontSize": "15px",
              "marginLeft": "3px",
              "padding": "3px 5px",
            }
          }
        >
          <i
            className="material-icons"
          >
            person_pi
          </i>
        </li>
        <li>
          <a
            href="#/"
            onClick={[Function]}
          >
            Logout
          </a>
        </li>
        <li>
          <a
            href="#/sources"
            onClick={[Function]}
          >
            Sources
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Navigation Links  Should display Logout 1`] = `
<nav>
  <div
    className=""
  >
    <div
      className="navBar"
    >
      <div
        className=""
      >
        <a
          className="brand-logo center"
          href="#/"
          onClick={[Function]}
        >
          Articles Hub
        </a>
      </div>
      <a
        className="button-collapse"
        data-activates="mobile-demo"
        href="#"
      >
        <i
          className="material-icons"
        >
          toc
        </i>
      </a>
      <ul
        className="left hide-on-med-and-down"
        id="nav-mobile"
      >
        <li>
          <a
            href="#/sources"
            onClick={[Function]}
          >
            Sources
          </a>
        </li>
      </ul>
      <ul
        className="right hide-on-med-and-down"
        id="nav-mobile"
      >
        <li>
          <a
            href="#"
          />
        </li>
        <li>
          <a
            href="#/"
            onClick={[Function]}
          >
            Logout
          </a>
        </li>
      </ul>
    </div>
    <div>
      <ul
        className="side-nav"
        id="mobile-demo"
        style={
          Object {
            "closeOnClick": true,
            "color": "#6D29C5",
            "draggable": true,
            "edge": "right",
            "fontSize": "16px",
            "marginLeft": "3px",
            "padding": "3px 5px",
          }
        }
      >
        <li
          style={
            Object {
              "color": "#6D29C5",
              "fontSize": "15px",
              "marginLeft": "3px",
              "padding": "3px 5px",
            }
          }
        >
          <i
            className="material-icons"
          >
            person_pi
          </i>
        </li>
        <li>
          <a
            href="#/"
            onClick={[Function]}
          >
            Logout
          </a>
        </li>
        <li>
          <a
            href="#/sources"
            onClick={[Function]}
          >
            Sources
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
